model_api/cpp/models/src/segmentation_model.cpp

```diff
@@ -217,41 +259,43 @@ void SegmentationModel::prepareInputsOutputs(std::shared_ptr<ov::Model>& model)
-            }
-            predictions.convertTo(result->resultImage, CV_8UC1);
-        } else if (outTensor.get_element_type() == ov::element::f32) {
-            const float* data = outTensor.data<float>();
-            for (int rowId = 0; rowId < outHeight; ++rowId) {
-                for (int colId = 0; colId < outWidth; ++colId) {
-                    int classId = 0;
-                    float maxProb = -1.0f;
-                    for (int chId = 0; chId < outChannels; ++chId) {
-                        float prob = data[chId * outHeight * outWidth + rowId * outWidth + colId];
-                        if (prob > maxProb) {
-                            classId = chId;
-                            maxProb = prob;
-                        }
-                    }  // nChannels
-
-                    result->resultImage.at<uint8_t>(rowId, colId) = classId;
-                }  // width
-            }  // height
+    cv::Mat soft_prediction;
+    if (outChannels == 1 && outTensor.get_element_type() == ov::element::i32) {
+        cv::Mat predictions(outHeight, outWidth, CV_32SC1, outTensor.data<int32_t>());
+        predictions.convertTo(soft_prediction, CV_8UC1);
+    } else if (outChannels == 1 && outTensor.get_element_type() == ov::element::i64) {
+        cv::Mat predictions(outHeight, outWidth, CV_32SC1);
+        const auto data = outTensor.data<int64_t>();
+        for (size_t i = 0; i < predictions.total(); ++i) {
+            reinterpret_cast<int32_t*>(predictions.data)[i] = int32_t(data[i]);
+        }
+        predictions.convertTo(soft_prediction, CV_8UC1);
+    } else if (outTensor.get_element_type() == ov::element::f32) {
+        float* data = outTensor.data<float>();
+        std::vector<cv::Mat> channels;
+        for (size_t c = 0; c < outTensor.get_shape()[1]; ++c) {
+            channels.emplace_back(cv::Size{outHeight, outWidth}, CV_32F, data + c * outHeight * outWidth);
         }
+        cv::merge(channels, soft_prediction);
+        cv::resize(soft_prediction, soft_prediction, {inputImgSize.inputImgWidth, inputImgSize.inputImgHeight}, 0.0, 0.0, cv::INTER_NEAREST);
+    }
+    cv::Mat hard_prediction = create_hard_prediction_from_soft_prediction(soft_prediction, soft_threshold, blur_strength);
+    auto annotations = create_annotation_from_segmentation_map(hard_prediction, soft_prediction, labels);
+
+    if (return_soft_prediction) {
+        ImageResultWithSoftPrediction* result = new ImageResultWithSoftPrediction(infResult.frameId, infResult.metaData);
+        result->annotations = annotations;
+        result->resultImage = hard_prediction;
+        result->soft_prediction = soft_prediction;
+        return std::unique_ptr<ResultBase>(result);
+    } else {
+        ImageResult* result = new ImageResult(infResult.frameId, infResult.metaData);
+        result->annotations = annotations;
+        result->resultImage = cv::Mat(outHeight, outWidth, CV_8UC1);
 
         cv::resize(result->resultImage,
```

Comment:
You just resize uninitialized cv::Mat here because you have `result->resultImage = cv::Mat(outHeight, outWidth, CV_8UC1);`

Reply:
Fixed.